File: TestApi/Model/views/ProjectModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ApiServer.Model.views
{
    public static class ProjectModel
    {
        public class ProjectForView 
        {
            public Guid Id { get; set; }
            public string Name { get; set; }
            public int TotalImg { get; set; }
            public int TotalImgNotTagged { get; set; }
            public int TotalImgNotClassed { get; set; }
            public int TotalImgQC { get; set; }
            public int TotalImgNotQC { get; set; }
            public bool IsDisabled { get; set; }
            public string Description { get; set; }
            public string Note { get; set; }
            public string Usernames { get; set; }
        }

        public class ProjectForCreate {
            public int Id { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public string Note { get; set; }
            
        }

        public class ProjectForUpdate
        {
            public Guid Id { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public string Note { get; set; }

        }
    }
}

File: Client/src/app/shared/models/project.model.ts
export interface IProject{
    id: string;
    name: string;
    description: string;
}

export class ProjectForView implements IProject{
    constructor(public id:string = null,public name: string = null,public description: string = null, public note: string = null, public totalImg: number = 0,public totalImgNotTagged: number = 0, public totalImgNotClassed: number = 0, public totalImgQC: number = 0, public totalImgNotQC: number = 0, public usernames: string = null, public isDisable: boolean = false){}
}

export class ProjectForAdd implements IProject{
    constructor(public id:string,public name: string,public description: string = null, public note: string = null){}
}

export class ProjectForUpdate{
    constructor(public id:string,public name: string,public description: string = null, public note: string = null){}
}